lib/model/game_state.dart

```diff
@@ -35,6 +38,29 @@ class GameState with ChangeNotifier {
       discard.add(currentCard);
     }
 
+    // TODO - Game Result
+    switch (result){
+      case GameResult.WIN: {
+
+      }
+      break;
+
+      case GameResult.LOSE: {
+
+      }
+      break;
+
+    }
+
+    // Check special card conditions
+    if (checkMutiny(playerHand)){
+      deck.insert(0, Mutiny());
+    }else if (checkNavyRaid(playerHand)){
+      deck.insert(0, NavyRaid());
+    }else if (checkRavenousCrew(playerHand)){
+      deck.insert(0, RavenousCrew());
+    };
```

Comment:
Thinking we could refactor this over into the `special_cards` file to stop this method from becoming too long (and also so all that logic is encapsulated in the one place). Something like `addSpecialTopCardToDeck(playerHand);` ?

Reply:
Agreed, done.